zio-http/src/main/scala/zio/http/URL.scala

```diff
@@ -98,6 +98,10 @@ final case class URL(
    * location.
    */
   def toJavaURL: Option[java.net.URL] = if (self.kind == URL.Location.Relative) None else Some(toJavaURI.toURL)
+
+  def queryParamsAsString: String =
+    Option(self.toJavaURI.getQuery).getOrElse("")
+
```

Comment:
I am not really convinced if this is the right way to expose query params.
We should ideally have a type such as - `QueryParam` that has methods such as `add`, `remove`, `encode` etc.

The above API seems a bit to specific for a particular use-case where the default value is empty string and will introduce an unnecessary breaking change later when something more ergonomic is available.

I think for now we can do this - 
```scala
url.toJavaURL.map(_.getQuery).getOrElse("")
```

Reply:
@tusharmath 
I pushed an alternative that I think satisfies your intention.

I think that `queryParamsAsString` supporting only java urls might be confusing. Someone may make a URL from a ws endpoint and this new impl would throw because ws endpoints are a malformed java urls.

I kept my initial implementation and renamed it `uriQueryParamsAsString`.
I added a `urlQueryParamsAsString` where I use your suggestion. The only tweaks are to make it return the query decoded. I also refactored `toJavaURL` to make it pure (let me know if this should be reverted)

Reply:
I do think it would be a good idea to have `def queryParams: QueryParams`, with methods like add/remove/etc.

Perhaps the optimal way to solve the problem would be `url.queryParams.encode`. 

Would you sign off on that API? Only a bit more typing for someone looking for a "fast" way to get the query parameters as an encoded string.

Reply:
Like this?
```
  trait QueryParams{
    def add(key: String, value: String): QueryParams
    def remove(key: String): QueryParams
    //this returns URI query params
    def queryParamsMap: Map[String, List[String]]
    //these return URI query params - or we can have a URLQueryParams and URIQuery params
    def decoded: String
    def encoded: String
  }
```

Also do we work on the URI level or want to make a distinction between URI and URL?